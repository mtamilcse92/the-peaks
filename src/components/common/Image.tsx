import React from "react";
import { ReactComponent as BrandImage } from "../../assets/brand-icon.svg";
import styles from "./Image.module.scss";

export type Props = {
  imageURL?: string;
};

export const DefaultImage = () => (
  <div className={styles.defaultImage}>
    <BrandImage className={styles.brandImage} />
  </div>
);

const Image: React.FC<Props> = ({ imageURL }) => {
  if (!imageURL) {
    return <DefaultImage />;
  }
  return <img src={imageURL} alt="Unknown" />;
};

export default Image;
