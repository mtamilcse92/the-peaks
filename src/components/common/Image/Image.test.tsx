import React from "react";
import { render } from "@testing-library/react";
import Image from "./index";

describe("<Image /> spec", () => {
  it("renders the Image", () => {
    const WithDefaultImage = render(
      <Image/>
    );
    const WithCustomImage = render(
      <Image
      imageURL="https://media.guim.co.uk/ecf0db0f2a9a036453cdaf9978b73bc8d7a733b4/17_0_1125_675/500.jpg"
      />
    );
    expect(WithCustomImage).toMatchSnapshot();
    expect(WithDefaultImage).toMatchSnapshot();
  });
});
