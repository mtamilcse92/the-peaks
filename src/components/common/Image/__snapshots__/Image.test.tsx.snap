// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> spec renders the Image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="defaultImage"
      >
        <svg
          class="brandImage"
        >
          brand-icon.svg
        </svg>
      </div>
    </div>
    <div>
      <img
        alt="Unknown"
        class="container"
        src="https://media.guim.co.uk/ecf0db0f2a9a036453cdaf9978b73bc8d7a733b4/17_0_1125_675/500.jpg"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Unknown"
      class="container"
      src="https://media.guim.co.uk/ecf0db0f2a9a036453cdaf9978b73bc8d7a733b4/17_0_1125_675/500.jpg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Image /> spec renders the Image 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="defaultImage"
      >
        <svg
          class="brandImage"
        >
          brand-icon.svg
        </svg>
      </div>
    </div>
    <div>
      <img
        alt="Unknown"
        class="container"
        src="https://media.guim.co.uk/ecf0db0f2a9a036453cdaf9978b73bc8d7a733b4/17_0_1125_675/500.jpg"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="defaultImage"
    >
      <svg
        class="brandImage"
      >
        brand-icon.svg
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
